client/state/data-layer/wpcom/sites/utils.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import {
+	COMMENTS_REMOVE,
+	COMMENTS_RECEIVE,
+	COMMENTS_COUNT_INCREMENT,
+} from 'state/action-types';
+import { getSitePost } from 'state/posts/selectors';
+import { errorNotice } from 'state/notices/actions';
+
+/***
+ * Creates a placeholder comment for a given text and postId
+ * We need placehodler id to be unique in the context of siteId, postId for that specific user,
+ * date milliseconds will do for that purpose.
+ *
+ * @param   {String}           commentText     text of the comment
+ * @param   {Number}           postId          post identifier
+ * @param   {Number|undefined} parentCommentId parent comment identifier
+ * @returns {Object}                           comment placeholder
+ */
+export const createPlaceholderComment = ( commentText, postId, parentCommentId ) => ( {
+	ID: 'placeholder-' + ( new Date().getTime() ),
+	parent: parentCommentId ? { ID: parentCommentId } : false,
+	date: ( new Date() ).toISOString(),
+	content: commentText,
+	status: 'pending',
+	type: 'comment',
+	post: { ID: postId },
+	isPlaceholder: true,
+	placeholderState: 'PENDING'
+} );
+
+/***
+ * updates the placeholder comments with server values
+ *
+ * @param {Function} dispatch redux dispatcher
+ * @param {Object}   action   redux action
+ * @param {Function} next     dispatches to next middleware in chain
+ * @param {Object}   comment  updated comment from the request response
+ */
+export const updatePlaceholderComment = ( { dispatch }, { siteId, postId, parentCommentId, placeholderId }, next, comment ) => {
+	// remove placeholder from state
+	dispatch( { type: COMMENTS_REMOVE, siteId, postId, commentId: placeholderId } );
+	// add new comment to state with updated values from server
+	dispatch( { type: COMMENTS_RECEIVE, siteId, postId, comments: [ comment ], skipSort: !! parentCommentId } );
+	// increment comments count
+	dispatch( { type: COMMENTS_COUNT_INCREMENT, siteId, postId } );
```

Comment:
why are we issuing this action? it seems like the comment count reducer should be watching for `COMMENTS_REMOVE` and `COMMENTS_RECEIVE` instead…